academy/w3bstream-dimo-guide.md

```diff
@@ -63,21 +64,52 @@ In the context of our example, the registration process includes the following a
 2. Register each distinct device ID within a designated registry contract on the IoTeX blockchain.
 3. Obtain the user's wallet address through Metamask login and generate binding records within the binding registry contract. These records solidify the connection between each device ID owned by the user and their respective wallet address.
 
-![Screenshot of the registration process]
-
 For detailed insights, the Device Registry, Device Binding, and SBT Token contracts can be located within the blockchain hardhat project directory of the project repository.
 
-[Inclusion of relevant code snippets showcasing the device registration, binding, and minting processes]
+The following snippet of code showes how the Device Registry contract gets called to register a new device id on the blockchain:
+```Typescript
+// adapter/features/web3/services/viem/registrationContract.ts
+
+export async function registerDevice(deviceIds: string[]) {
+
+  /* ... */
+
+  const { request } = await publicClient.simulateContract({
+    account: walletClient.account,
+    address: registryConfig.address as `0x${string}`,
+    abi: registryConfig.abi,
+    functionName: "registerDevices",
+    args: [devicesToRegister],
+  });
+  const hash = await walletClient.writeContract(request);
+  return publicClient.waitForTransactionReceipt({ hash, confirmations: 1 });
+}
+```
 
 ## Data fetching
 
 Once a user has successfully registered their DIMO devices within our project, they can log in to the web application providing their DIMO API Token. (Note: In an ideal scenario, further enhancement could be achieved by incorporating an authentication mechanism such as OAuth 2.0, if supported by the utilized API).
 
 Ideally, our web application would securely store the user's API token, orchestrating periodic data retrieval and seamless forwarding to our W3bstream logic. To provide clarity and facilitate the step-by-step process, we've introduced a dedicated "Sync Data" page. Upon logging in, users initiate a request to fetch the latest driving data. This acquired data is then seamlessly channeled into the W3bstream project, where it undergoes meticulous processing and analysis.
 
-[Display relevant code snippets illustrating DIMO authentication, data fetching, and subsequent transmission to W3bstream.]
+The following code snippets shows the DIMO client code that fetches the device data:
+```typescript
+// adapter/features/data/services/dimo/client.ts
+
+/* ... */
+
+// Fetch DIMO device data 
+export const dimoDeviceDataAxiosInstance = (token: string) =>
```

Comment:
this part only creates an axios instance without fetching. We then use this instance to fetch data with url path and body

Reply:
Thanks, I've added the additional snippet.
I've also improved the layout of the images